engine/src/flutter/shell/platform/embedder/embedder_semantics_update.cc

```diff
@@ -126,6 +126,44 @@ FlutterSemanticsFlag SemanticsFlagsToInt(const SemanticsFlags& flags) {
   return static_cast<FlutterSemanticsFlag>(result);
 }
 
+const FlutterSemanticsFlags convertToFlutterSemanticsFlags(
+    const SemanticsFlags& source) {
+  FlutterSemanticsFlags dest = {};
+
+  dest.hasCheckedState = source.hasCheckedState;
+  dest.isChecked = source.isChecked;
+  dest.isSelected = source.isSelected;
+  dest.isButton = source.isButton;
+  dest.isTextField = source.isTextField;
+  dest.isFocused = source.isFocused;
+  dest.hasEnabledState = source.hasEnabledState;
+  dest.isEnabled = source.isEnabled;
+  dest.isInMutuallyExclusiveGroup = source.isInMutuallyExclusiveGroup;
+  dest.isHeader = source.isHeader;
+  dest.isObscured = source.isObscured;
+  dest.scopesRoute = source.scopesRoute;
+  dest.namesRoute = source.namesRoute;
+  dest.isHidden = source.isHidden;
+  dest.isImage = source.isImage;
+  dest.isLiveRegion = source.isLiveRegion;
+  dest.hasToggledState = source.hasToggledState;
+  dest.isToggled = source.isToggled;
+  dest.hasImplicitScrolling = source.hasImplicitScrolling;
+  dest.isMultiline = source.isMultiline;
+  dest.isReadOnly = source.isReadOnly;
+  dest.isFocusable = source.isFocusable;
+  dest.isLink = source.isLink;
+  dest.isSlider = source.isSlider;
+  dest.isKeyboardKey = source.isKeyboardKey;
+  dest.isCheckStateMixed = source.isCheckStateMixed;
+  dest.hasExpandedState = source.hasExpandedState;
+  dest.isExpanded = source.isExpanded;
+  dest.hasSelectedState = source.hasSelectedState;
+  dest.hasRequiredState = source.hasRequiredState;
+  dest.isRequired = source.isRequired;
+
+  return dest;
```

Comment:
Instead, use designated initialiser syntax:
```cc
const FlutterSemanticsFlags convertToFlutterSemanticsFlags(
    const SemanticsFlags& source) {
  return {
    .has_checked_state = source.hasCheckedState,
    .is_checked = source.isChecked,
    // ...
    .is_required = source.isRequired,
  };
}
```

I imagine in both cases the compiler is clever enough to apply return value optimisation, which ensures the return object is constructed directly in the memory address of the caller rather than copied, but this way is more idiomatic and avoids declaring a local at all.